libs/knowledge-store/ragstack_knowledge_store/cassandra.py

```diff
@@ -447,8 +448,13 @@ def _linked_ids(
         results = self._session.execute(self._query_linked_ids, (source_id,))
         return _results_to_ids(results)
 
-    def retrieve(
-        self, query: Union[str, Iterable[str]], *, k: int = 4, depth: int = 1
+    def traversing_retrieve(
+        self,
+        query: Union[str, Iterable[str]],
```

Comment:
I think for simplicity / consistency we could probably have this take `query: str`. Thinking about it, it's going to be difficult for more sophisticated retrievers to support the union directly, and that should probably be implemented as part of the things that will actually need to take multiple queries.

Reply:
done